app/views/govuk_publishing_components/components/_document_list.html.erb

```diff
@@ -17,14 +17,28 @@
 
   brand ||= false
   brand_helper = GovukPublishingComponents::AppHelpers::BrandHelper.new(brand)
+
+  li_data_attributes ||= {}
+
+  disable_ga4 ||= false
```

Comment:
I'd reserve the `disable_ga4` option for components where the tracking is entirely handled automatically. For situations like this, where the tracking only works if attributes are passed, would it make more sense to leave out this option but make the developer include the right module in the passed data attributes?

Reply:
(similar to the approach taken for UA tracking)

Reply:
@andysellick Sorry for the confusion, here the tracking here can still work when no attributes are passed through to the component. The `ga4_data` is for any extra GA4 values can be added via the data attributes - maybe I should have called it `extra_ga4_data` instead (I can update this name for the share links tracking too if preferred.) I will be using this attribute to pass the `section` value down to the tracking.

Though I'm still happy to remove the tracking from the component and pass it through as data attributes instead, if you prefer that?

Reply:
Thanks, I've renamed it to `ga4_extra_data`

Reply:
Nice, thanks, I see you've made the corresponding change on the share links component as well 👍